posit_workbench_best_practice_with_r.md

```diff
@@ -0,0 +1,485 @@
+```
+
+and we want to approximate the area of the sepal and petal by multiplying "Length" × "Width", and convert the Species values to upper-case.
+
+When working with standard R methods, we could modify the existing iris by adding and modifying columns.
+
+```r
+data("iris")
+iris$Sepal.Area <- iris$Sepal.Length * iris$Sepal.Width
+iris$Petal.Area <- iris$Petal.Length * iris$Petal.Width
+iris$Species <- toupper(iris$Species)
+head(iris)
+```
+
+```
+  Sepal.Length Sepal.Width Petal.Length Petal.Width Species Sepal.Area
+1          5.1         3.5          1.4         0.2  SETOSA      17.85
+2          4.9         3.0          1.4         0.2  SETOSA      14.70
+3          4.7         3.2          1.3         0.2  SETOSA      15.04
+4          4.6         3.1          1.5         0.2  SETOSA      14.26
+5          5.0         3.6          1.4         0.2  SETOSA      18.00
+6          5.4         3.9          1.7         0.4  SETOSA      21.06
+  Petal.Area
+1       0.28
+2       0.28
+3       0.26
+4       0.30
+5       0.28
+6       0.68
+```
+
+Meanwhile, using tidyverse methods, we can use mutate to create the modified table and overwrite the original iris table:
+
+```r
+library("dplyr")
+data("iris")
+iris <- mutate(iris,
+               Sepal.Area = Sepal.Length * Sepal.Width,
+               Petal.Area = Petal.Length * Petal.Width,
+               Species = toupper(Species))
+head(iris)
+```
+
+```
+  Sepal.Length Sepal.Width Petal.Length Petal.Width Species Sepal.Area
+1          5.1         3.5          1.4         0.2  SETOSA      17.85
+2          4.9         3.0          1.4         0.2  SETOSA      14.70
+3          4.7         3.2          1.3         0.2  SETOSA      15.04
+4          4.6         3.1          1.5         0.2  SETOSA      14.26
+5          5.0         3.6          1.4         0.2  SETOSA      18.00
+6          5.4         3.9          1.7         0.4  SETOSA      21.06
+  Petal.Area
+1       0.28
+2       0.28
+3       0.26
+4       0.30
+5       0.28
+6       0.68
+```
+
+This overwriting action prevents unnecessary redundant datasets being stored. But care must be taken as the overwriting cannot be undone. If you accidentally overwrite some data you needed, you'll need to recreate it again using your R script.
+
+### Avoiding intermediate variables
+
+It's common to perform a sequence of steps when preparing your data. This can lead to a series of intermediate variables being created to store the result of each step. Overwriting the results from the last step is one possible solution, but an alternative is to _pipe_ the result from one stage directly into the next.
+
+The [{magrittr}](https://magrittr.tidyverse.org/) package provides a _pipe_ operator, which is written in R as `%>%`. It takes the object to its left and "pipes" it into the function on the right as its first unset argument. In general terms, the following two lines of code are equivalent ways of running a function called `f`:
```

Comment:
Just highlighting that this is made available without explicitly calling {magrittr}. Also, should be mention the new pipe option, `|>`, as we'll have a more up to date version of R available?
```suggestion
The [{magrittr}](https://magrittr.tidyverse.org/) package provides a _pipe_ operator, which is written in R as `%>%`. This is also made available through other packages, such as [{dplyr}]((https://dplyr.tidyverse.org/). It takes the object to its left and "pipes" it into the function on the right as its first unset argument. In general terms, the following two lines of code are equivalent ways of running a function called `f`:
```